src/main/java/commands/ListDueCommand.java

```diff
@@ -34,14 +31,28 @@ private void loadAllChapters(Storage storage, Ui ui) {
     public void execute(Ui ui, Access access, Storage storage) {
         loadAllChapters(storage, ui);
 
-        for (Chapter chapter : allChapters) {
-            if (Scheduler.isDeadlineDue(chapter.getDueBy())) {
+        for (DueChapter chapter : allChapters) {
+            LocalDate deadline = chapter.getChapter().getDueBy();
+            if (Scheduler.isDeadlineDue(deadline)) {
                 dueChapters.add(chapter);
             }
         }
-        ui.showToUser("The chapters you have due by today are:");
-        for (Chapter dueChapter : dueChapters) {
-            ui.showToUser(dueChapter.toString());
+        if (dueChapters.size() == 0) {
+            ui.showToUser("You have no tasks due today! Please check back again tomorrow!");
+        } else {
```

Comment:
I think for this, you can have a return after line 41 so that you would not need the else part. Then the rest of the code will follow. Something like this:
if (dueChapters.size() == 0) {
    ui.showToUser("You have no tasks due today! Please check back again tomorrow!");
    return;
}
// rest of the code